File: csharp/wordy/Wordy.cs
using System;
using System.Text.RegularExpressions;

public static class Wordy
{
    public static int Answer(string question)
    {
        var expression = ParseExpression(question);
        return SolveExpression(expression);
    }

    private static string ParseExpression(string question)
    {
        const string pattern = @"^-?\d+(?: (?:(plus|minus|multiplied by|divided by) -?\d+))*$";
        
        var expression = question
            .Replace("What is ", "")
            .TrimEnd('?');

        if (!Regex.IsMatch(expression, pattern))
        {
            throw new ArgumentException("Invalid expression");
        }

        return expression
            .Replace("plus", "+")
            .Replace("minus", "-")
            .Replace("multiplied by", "*")
            .Replace("divided by", "/");
    }

    private static int SolveExpression(string expression)
    {
        var parts = expression.Split(" ");
        var lhs = int.Parse(parts[0]);
        
        for (int i = 1; i < parts.Length; i += 2)
        {
            var rhs = int.Parse(parts[i + 1]);
            var result = parts[i] switch
            {
                "+" => lhs + rhs,
                "-" => lhs - rhs,
                "*" => lhs * rhs,
                "/" => lhs / rhs,
                _ => throw new ArgumentException($"Invalid operator in expression {expression}")
            };

            lhs = result;
        }

        return lhs;
    }
}

File: kotlin/wordy/src/main/kotlin/Wordy.kt
import kotlin.math.pow

object Wordy {
    private val mathQuestion = Regex("""^What is (.*)\?$""")

    fun answer(input: String): Int {
        val question = mathQuestion.find(input) ?: throw UnsupportedOperationException("Not a math question: $input")

        val expression = question.groupValues.last()
            .replace("plus", "+")
            .replace("minus", "-")
            .replace("multiplied by", "*")
            .replace("divided by", "/")
            .replace(Regex("""([-\d]+) raised to the (\d+)[a-z]+ power"""), "$1 ^ $2")
            .split(" ")

        return solve(expression)
    }

    private fun solve(expression: List<String>): Int {
        if (expression.size == 1) {
            return expression[0].toInt()
        }

        require(expression.size >= 3) { "Invalid expression: $expression" }

        return expression
            .drop(1)
            .chunked(2) { Pair(it[0], it[1].toInt())}
            .fold(expression[0].toInt()) { lhs, (op, rhs) ->
                when (op) {
                    "+" -> lhs + rhs
                    "-" -> lhs - rhs
                    "*" -> lhs * rhs
                    "/" -> lhs / rhs
                    "^" -> lhs.toDouble().pow(rhs).toInt()
                    else -> throw UnsupportedOperationException("Unknown operator: $op")
                }
            }
    }
}
